lib/membrane_h264_plugin/parser.ex

```diff
@@ -319,40 +327,62 @@ defmodule Membrane.H264.Parser do
             output_alignment: state.output_alignment
           )
 
-        {[stream_format: {:output, fmt}], fmt.profile}
+        max_frame_reorder = if fmt.profile in [:baseline, :constrained_baseline], do: 0, else: 15
```

Comment:
As discussed, that can cause some problems when the profile is changed during the stream run